spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/HttpClientProperties.java

```diff
@@ -293,6 +296,64 @@ public String toString() {
 		/** The default ssl configuration type. Defaults to TCP. */
 		private SslProvider.DefaultConfigurationType defaultConfigurationType = SslProvider.DefaultConfigurationType.TCP;
 
+		/** Spring Cloud Gateway keystore path. */
+		private String keyStore;
+
+		/** Spring Cloud Gateway keystore type, default is JKS. */
+		private String keyStoreType;
+
+		/** Spring Cloud Gateway keystore provider, optional field. */
+		private String keyStoreProvider;
+
+		/** Spring Cloud Gateway keystore password. */
+		private String keyStorePassword;
+
+		/**
+		 * Spring Cloud Gateway key password, if not passed will consider same as
```

Comment:
"key password. If..."

Reply:
Also, why does this field exist? It is only ever used if keyStorePassword is not set.

Reply:
It might be possible that keystore password and key password are not the same, in that case, both passwords are needed.